Automattic/wp-calypso
# Plans 2023: Update action buttons to Upgrade when site on shorter term plan

Fixes https://github.com/Automattic/wp-calypso/issues/74637

## Proposed Changes

Updates the plan action buttons to direct towards an "Upgrade" when the site plan is on a shorter term. This applies to all `n-yearly` grid variations:  `/plans/yearly`, `/plans/2yearly`, `/plans/3yearly`. Button texts should read "Upgrade to Yearly", "Upgrade to Biannual", and "Upgrade to Triannual", given a respective site with a lower-term plan.

### Media

- `/plans/yearly/[SITE]` with site on Premium **monthly**:

<img width="700" alt="Screenshot 2023-03-22 at 3 39 14 PM" src="https://user-images.githubusercontent.com/1705499/226922458-76609246-a9a7-4e92-b558-36774e9a24db.png">

- `/plans/2yearly/[SITE]` with site on Premium **monthly**:

<img width="700" alt="Screenshot 2023-03-22 at 3 39 32 PM" src="https://user-images.githubusercontent.com/1705499/226922516-591c9bb7-623f-46a9-93e9-2d674c242ba2.png">

- `/plans/3yearly/[SITE]` with site on Premium **monthly**:

<img width="700" alt="Screenshot 2023-03-22 at 3 39 45 PM" src="https://user-images.githubusercontent.com/1705499/226922568-8b7040b9-8a64-4cb9-9927-e12724ec475d.png">

- `/plans/yearly/[SITE]` with site on Business **yearly**:

<img width="700" alt="Screenshot 2023-03-22 at 3 40 00 PM" src="https://user-images.githubusercontent.com/1705499/226922617-c56ad739-2535-420c-9a11-bb0ff72c0add.png">


## Testing Instructions



* Go to each of `/plans/monthly`, `/plans/yearly`, `/plans/2yearly`, `/plans/3yearly` with a variation of site plans (monthly, yearly, biannual, triannual)
* Confirm the updated button texts for the respective longer-term plan variant, as in the media ("Upgrade to ...")
* Confirm the other buttons display and work as expected (downgrade, upgrade, manage-plan, etc.)
* Confirm clicking on "Upgrade to ..." goes to checkout with the respective plan in the cart

## Pre-merge Checklist
